scripts/eurostat/health_determinants/common/euro_stat.py

```diff
@@ -0,0 +1,343 @@
+method needs to implemented by Subclasses.
+"""
+import os
+import sys
+import re
+import pandas as pd
+import numpy as np
+from absl import flags
+
+# For import common.replacement_functions
+_COMMON_PATH = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.insert(1, _COMMON_PATH)
+# pylint: disable=wrong-import-position
+from common.replacement_functions import (split_column, replace_col_values)
+from common.sv_config import file_to_sv_mapping
+
+# For import util.alpha2_to_dcid
+_COMMON_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), '../../../../'))
+sys.path.insert(1, _COMMON_PATH)
+# pylint: disable=wrong-import-order
+from util.alpha2_to_dcid import COUNTRY_MAP
+# pylint: enable=wrong-import-position
+# pylint: enable=wrong-import-order
+
+_FLAGS = flags.FLAGS
+default_input_path = os.path.join(os.path.dirname(os.path.abspath(__file__)),
+                                  "input_files")
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+_TMCF_TEMPLATE = ("Node: E:eurostat_population_{import_name}->E0\n"
+                  "typeOf: dcs:StatVarObservation\n"
+                  "variableMeasured: C:eurostat_population_{import_name}->SV\n"
+                  "measurementMethod: C:eurostat_population_{import_name}->"
+                  "Measurement_Method\n"
+                  "observationAbout: C:eurostat_population_{import_name}->geo\n"
+                  "observationDate: C:eurostat_population_{import_name}->time\n"
+                  "scalingFactor: 100\n"
+                  "value: C:eurostat_population_{import_name}->observation\n")
+
+
+class EuroStat:
+    """
+    EuroStat is a base class which provides common implementation for generating
+    CSV, MCF and tMCF files.
+    """
+    # Below variables will be initialized by sub-class (import specific)
+    _import_name = ""
+    _mcf_template = ""
+    _sv_value_to_property_mapping = {}
+    _sv_properties_template = {}
+    _sv_properties = {}
+
+    def __init__(self,
+                 input_files: list,
+                 csv_file_path: str = None,
+                 mcf_file_path: str = None,
+                 tmcf_file_path: str = None) -> None:
+        self._input_files = input_files
+        self._cleaned_csv_file_path = csv_file_path
+        self._mcf_file_path = mcf_file_path
+        self._tmcf_file_path = tmcf_file_path
+        self._df = pd.DataFrame()
+
+    # pylint: disable=pointless-statement
+    # pylint: disable=no-self-use
+    # pylint: disable=unused-argument
+
+    def set_cleansed_csv_file_path(self, cleansed_csv_file_path: str) -> None:
+        self._cleaned_csv_file_path = cleansed_csv_file_path
+
+    def set_mcf_file_path(self, mcf_file_path: str) -> None:
+        self._mcf_file_path = mcf_file_path
+
+    def set_tmcf_file_path(self, tmcf_file_path: str) -> None:
+        self._tmcf_file_path = tmcf_file_path
+
+    def _property_correction(self):
```

Comment:
Pls add a comment indicating what child class overriding these methods are expected to do and what variables/state is it likely modify.